src/app/modules/item/components/item-remove-button/item-remove-button.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { ReplaySubject, combineLatest } from 'rxjs';
+import { distinct, map, switchMap, filter } from 'rxjs/operators';
+import { mapToFetchState } from '../../../../shared/operators/state';
+import { GetItemChildrenService, ItemChild } from '../../http-services/get-item-children.service';
+import { Item } from '../../http-services/get-item-by-id.service';
+import { ItemDataSource } from '../../services/item-datasource.service';
+import { isNotNullOrUndefined } from '../../../../shared/helpers/null-undefined-predicates';
+import { ConfirmationService } from 'primeng/api';
+import { RemoveItemService } from '../../http-services/remove-item.service';
+import { ActionFeedbackService } from '../../../../shared/services/action-feedback.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'alg-item-remove-button',
+  templateUrl: './item-remove-button.component.html',
+  styleUrls: [ './item-remove-button.component.scss' ],
+})
+export class ItemRemoveButtonComponent implements OnChanges {
+  @Input() item?: Item;
+
+  private readonly id$ = new ReplaySubject<string>(1);
+  readonly state$ = combineLatest([
+    this.id$,
+    this.itemDataSource.state$.pipe(
+      map(state => {
+        if (state.isReady) {
+          return state.data.route.attemptId || state.data.route.parentAttemptId;
+        }
+        return null;
+      }),
```

Comment:
the item datasource service should only be injected in page components, if you need the route or anything else, you can pass `itemData` as input instead of just an item

Reply:
Fixed

Reply:
Fixed